eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -567,11 +585,95 @@ public function locationCopied($locationId, $newLocationId, $newParentId)
         $this->copySubtree(
             $actionMap,
             $oldParentAliasId,
-            $newParentAliasId,
-            $locationId
+            $newParentAliasId
         );
     }
 
+    public function locationSwapped($location1Id, $location1ParentId, $location2Id, $location2ParentId)
+    {
+        $location1Entries = $this->gateway->loadLocationEntries($location1Id);
+        $location2Entries = $this->gateway->loadLocationEntries($location2Id);
+
+        $location1MainLanguageId = $this->gateway->getLocationContentMainLanguageId($location1Id);
+        $location2MainLanguageId = $this->gateway->getLocationContentMainLanguageId($location2Id);
+
+        // Load autogenerated entries to find alias ID
+        $autoLocation1 = $this->gateway->loadAutogeneratedEntry("eznode:{$location1Id}");
+        $autoLocation2 = $this->gateway->loadAutogeneratedEntry("eznode:{$location2Id}");
+
+        // Historize first, in case swapped Locations are siblings.
+        // We need to historize everything separately per language (mask), in case the entries
+        // remain history future publishing reusages need to be able to take them over cleanly.
```

Comment:
Improved in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/8f1bf00eaa60baaa3d8f85908e39708f44c5c9e5